src/Workspaces/Remote/ServiceHub/Services/DiagnosticAnalyzer/DiagnosticComputer.cs

```diff
@@ -37,6 +37,38 @@ internal class DiagnosticComputer
         /// pressure when there are large number of open documents across different projects to be analyzed by background analysis.
         /// </summary>
         private static CompilationWithAnalyzersCacheEntry? s_compilationWithAnalyzersCache = null;
+
+        /// <summary>
+        /// List of high priority diagnostic computation tasks which are currently executing.
+        /// Any new high priority diagnostic request is added to this list before the core diagnostics
+        /// compute call is performed, and removed from this list after the computation finishes.
+        /// Any new normal priority diagnostic request first waits for all the high priority tasks in this list
+        /// to complete, and moves ahead only after this list becomes empty.
+        /// </summary>
+        /// <remarks>
+        /// Read/write access to the list is guarded by <see cref="s_gate"/>.
+        /// </remarks>
+        private static readonly List<Task> s_highPriorityComputeTasks = new();
```

Comment:
nit: make this a set.  All you do is Add/Remove, and you want to assert about containment, and have everything be O(1).

Reply:
also, if this is an ImmutableHashSet, then you dont' need to make copies of it under the locks.  Just use the lock to do the atomic assignment to the field, but otherwise just use normal ops.

Reply:
Switched to an ImmutableHashSet